panelapp/panelapp_utils.py

```diff
@@ -0,0 +1,160 @@
+    del panel_gene_json['gene_data']  # No need to keep gene info as Seqr already maintain info
+    result = PaLocusListGene(
+        seqr_locus_list_gene=seqr_locus_list_gene,
+        confidence_level=panel_gene_json.get('confidence_level'),
+        biotype=panel_gene_json.get('biotype') or None,
+        penetrance=panel_gene_json.get('penetrance') or None,
+        mode_of_pathogenicity=panel_gene_json.get('mode_of_pathogenicity') or None,
+        mode_of_inheritance=panel_gene_json.get('mode_of_inheritance') or None,
+        raw_data=panel_gene_json,
+    )
+
+    return result
+
+
+def _get_all_panels(panels_url, all_results):
+    resp = requests.get(panels_url)
+    resp_json = resp.json()
+    curr_page_results = [r for r in resp_json.get('results', []) if r.get('stats', {}).get('number_of_genes', 0) > 0]
+    all_results += curr_page_results
+
+    next_page = resp_json.get('next', None)
+    if next_page is None:
+        return all_results
+    else:
+        return _get_all_panels(next_page, all_results)
+
+
+def _get_all_genes_for_panel(panel_genes_url, all_results):
+    resp = requests.get(panel_genes_url)
+    resp_json = resp.json()
+    all_results += resp_json.get('results', [])
+
+    next_page = resp_json.get('next', None)
+    if next_page is None:
+        return all_results
+    else:
+        return _get_all_genes_for_panel(next_page, all_results)
+
+
+def _create_or_update_locus_list_from_panel(user, panelgenes_url, panel_json):
+    panel_app_id = panel_json.get('id')
+    existing = _safe_get_locus_list(panel_app_id)
+
+    name = panel_json['name']
+    disease_group = panel_json.get('disease_group') or None
+    disease_sub_group = panel_json.get('disease_sub_group') or None
+    status = panel_json.get('status') or None
+    version = panel_json.get('version') or None
+    version_created = panel_json.get('version_created') or None
+    description = _create_panel_description(panel_app_id, version, disease_group, disease_sub_group)
+    new_seqrlocuslist_json = {
+        'name': name,
+        'description': description,
+        'is_public': True,
+    }
+    new_palocuslist_json = {
+        'disease_group': disease_group,
+        'disease_sub_group': disease_sub_group,
+        'status': status,
+        'version': version,
+        'version_created': version_created,
+        'url': panelgenes_url,
+        'raw_data': panel_json,
+    }
+    if existing:
+        update_model_from_json(existing.seqr_locus_list, new_seqrlocuslist_json, user)
+        update_model_from_json(existing, new_palocuslist_json, user)
+
+        return existing
+    else:
+        seqr_locus_list = create_model_from_json(SeqrLocusList, new_seqrlocuslist_json, user)
+        pa_locus_list = PaLocusList.objects.create(seqr_locus_list=seqr_locus_list, panel_app_id=panel_app_id)
+        update_model_from_json(pa_locus_list, new_palocuslist_json, user)
+
+        return pa_locus_list
```

Comment:
totally okay to leave it as is, but I think it would be more readable to not duplicate the parts of the logic that are the same whether or not there was an existing list - 
```
pa_locus_list = _safe_get_locus_list(panel_app_id)
...
if pa_locus_list:
    update_model_from_json(existing.seqr_locus_list, new_seqrlocuslist_json, user)
else:
    seqr_locus_list = create_model_from_json(SeqrLocusList, new_seqrlocuslist_json, user)
    pa_locus_list = PaLocusList.objects.create(seqr_locus_list=seqr_locus_list, panel_app_id=panel_app_id)
update_model_from_json(pa_locus_list, new_palocuslist_json, user)
return pa_locus_list
```

Reply:
Agree your logic is more readable, changes applied.